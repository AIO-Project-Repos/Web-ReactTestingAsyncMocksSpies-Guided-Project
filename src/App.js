import React, { useState }  from 'react';
import logo from './logo.svg';
import './App.css';
import uuid from 'uuid'

jest.mock('uuid', () => {
  return () => 'hardcoded id'
});

// jest.mock('./helpers', () => {
//   return {
//     timestamp: () => 'same timestamp'
//   }
// })

export default function App() {
  const [count, setCount] = useState(0);

  const increaseCountSync = () => {
    setCount(count + 1);
  }

  const increaseCountAsync = () => {
    setTimeout(() => {
      setCount(count => count + 1);
    }, 200);
  }

  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <div>the count is {count}</div>

        <div>this is random: {uuid()}</div>

        <button onClick={increaseCountSync}>increase sync</button>
        <button onClick={increaseCountAsync}>increase async</button>
      </header>
    </div>
  );
}
