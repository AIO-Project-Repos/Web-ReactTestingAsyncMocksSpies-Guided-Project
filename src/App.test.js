import React from 'react';
import '@testing-library/jest-dom';
import * as rtl from '@testing-library/react';
import App from './App';

afterEach(rtl.cleanup);

let wrapper;

beforeEach(() => {
  wrapper = rtl.render(<App />);
});

it('matches snapshot', () => {
  expect(wrapper.container).toMatchSnapshot()
});
