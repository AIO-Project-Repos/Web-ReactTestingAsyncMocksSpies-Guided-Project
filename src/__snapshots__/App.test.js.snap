// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <img
        alt="logo"
        class="App-logo"
        src="logo.svg"
      />
      <div>
        the count is 
        0
      </div>
      <div>
        this is random: 
        hardcoded id
      </div>
      <button>
        increase sync
      </button>
      <button>
        increase async
      </button>
    </header>
  </div>
</div>
`;
